extraction/src/test/scala/org/broadinstitute/monster/encode/extraction/EncodeExtractionPipelineBuilderSpec.scala

```diff
@@ -0,0 +1,264 @@
+        ): Msg
+      }.to[mutable.ArrayBuffer]
+    )
+  )
+
+  // analysisStepVersions
+  val analysisStepVersionParams = ("@id", "1")
+
+  val analysisStepVersionOut = Obj(
+    Str("@graph") -> new Arr(
+      fakeIds.map { i =>
+        Obj(
+          Str("@id") -> Str(i.toString),
+          Str("analysis_step") -> Str(i.toString)
+        ): Msg
+      }.to[mutable.ArrayBuffer]
+    )
+  )
+
+  // analysisStep
+  val analysisStepParams = ("@id", "1")
+
+  val analysisStepOut = Obj(
+    Str("@graph") -> new Arr(
+      fakeIds.map { i =>
+        Obj(
+          Str("@id") -> Str(i.toString)
+        ): Msg
+      }.to[mutable.ArrayBuffer]
+    )
+  )
+
+  val responseMap = Map[(EncodeEntity, (String, String)), Msg](
+    (EncodeEntity.Biosample, biosampleParams) -> biosampleOut,
+    (EncodeEntity.Donor, donorParams) -> donorOut,
+    (EncodeEntity.Library, libraryParams) -> libraryOut,
+    (EncodeEntity.Replicate, replicateParams) -> replicateOut,
+    (EncodeEntity.AntibodyLot, antibodyParams) -> antibodyOut,
+    (EncodeEntity.Target, targetParams) -> targetOut,
+    (EncodeEntity.Experiment, experimentParams) -> experimentOut,
+    (EncodeEntity.FunctionalCharacterizationExperiment, fcExperimentParams) -> fcExperimentOut,
+    (EncodeEntity.File, fileParams) -> fileOut,
+    (EncodeEntity.AnalysisStepRun, analysisStepRunParams) -> analysisStepRunOut,
+    (EncodeEntity.AnalysisStepVersion, analysisStepVersionParams) -> analysisStepVersionOut,
+    (EncodeEntity.AnalysisStep, analysisStepParams) -> analysisStepOut
+  )
+
+  val mockClient = new MockEncodeClient(responseMap)
+}
+
+class EncodeExtractionPipelineBuilderSpec extends PipelineBuilderSpec[Args] {
+  import EncodeExtractionPipelineBuilderSpec._
+
+  val outputDir = File.newTemporaryDirectory()
+  override def afterAll(): Unit = outputDir.delete()
+
+  override val testArgs = Args(
+    outputDir = outputDir.pathAsString,
+    // batchSize is 2 so that the fcexperiment piece of the pipeline actually has something to interact with
+    batchSize = 2L
+  )
+
+  override val builder =
+    new ExtractionPipelineBuilder(getClient = () => mockClient)
+
+  behavior of "EncodeExtractionPipelineBuilder"
+
+  it should "query ENCODE in the expected way" in {
+    mockClient.recordedRequests.toSet shouldBe Set(
+      (EncodeEntity.Biosample, biosampleParams),
+      (EncodeEntity.Donor, donorParams),
+      (EncodeEntity.Library, libraryParams),
+      (EncodeEntity.Replicate, replicateParams),
+      (EncodeEntity.AntibodyLot, antibodyParams),
+      (EncodeEntity.Target, targetParams),
+      (EncodeEntity.Experiment, experimentParams),
+      (EncodeEntity.FunctionalCharacterizationExperiment, fcExperimentParams),
+      (EncodeEntity.File, fileParams),
+      (EncodeEntity.AnalysisStepRun, analysisStepRunParams),
+      (EncodeEntity.AnalysisStepVersion, analysisStepVersionParams),
+      (EncodeEntity.AnalysisStep, analysisStepParams)
+    )
+  }
```

Comment:
Can we structure these checks to be more fine-grained, with one `it should` per entity type? We can bundle the `recordedRequests` and `readMsgs` checks into the same cases